src/hidimstat/dcrt.py

```diff
@@ -1,307 +1,414 @@
-        selection_set = np.arange(n_features)
-
-    # Refit the model with the estimated support set
-    if refit and estimated_coef is None and selection_set.size < n_features:
-        clf_refit = clone(clf_screening)
-        clf_refit.fit(X_[:, selection_set], y_)
-        coef_X_full[selection_set] = np.ravel(clf_refit.coef_)
-
-    ## Distillation & calculate
-    if statistic == "residual":
-        # For distillation of X use least_square loss
-        results = Parallel(n_jobs, verbose=joblib_verbose)(
-            delayed(_lasso_distillation_residual)(
+
+    def __init__(
+        self,
+        estimated_coef=None,
+        sigma_X=None,
+        params_lasso_screening={
+            "alpha": None,
+            "n_alphas": 10,
+            "alphas": None,
+            "alpha_max_fraction": 0.5,
+            "cv": 5,
+            "tol": 1e-6,
+            "max_iter": 1000,
+            "fit_intercept": False,
+            "selection": "cyclic",
+        },
+        params_lasso_distillation_x=None,
+        params_lasso_distillation_y=None,
+        refit=False,
+        screening=True,
+        screening_threshold=1e1,
+        statistic="residual",
+        centered=True,
+        n_jobs=1,
+        joblib_verbose=0,
+        fit_y=False,
+        n_tree=100,
+        problem_type="regression",
+        random_state=2022,
+    ):
+        self.estimated_coef = estimated_coef
+        self.sigma_X = sigma_X
+        self.params_lasso_screening = params_lasso_screening
+        self.params_lasso_distillation_x = params_lasso_distillation_x
+        self.params_lasso_distillation_y = params_lasso_distillation_y
+        self.refit = refit
+        self.screening = screening
+        self.screening_threshold = screening_threshold
+        self.statistic = statistic
+        self.centered = centered
+        self.n_jobs = n_jobs
+        self.joblib_verbose = joblib_verbose
+        self.fit_y = fit_y
+        self.n_tree = n_tree
+        self.problem_type = problem_type
+        self.random_state = random_state
+
+    def fit(self, X, y):
+        """
+        Fit the dCRT model.
+        Based on the paper by :footcite:t:`liu2022fast` for fast conditional
+        randomization testing.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data matrix where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+
+        Notes
+        -----
+        The method follows these main steps:
+        1. Optional data centering using StandardScaler
+        2. Variable screening using LASSO (if not provided with estimated coefficients)
+        3. Feature selection based on coefficient magnitudes
+        4. Model refitting on selected features (if refit=True)
```

Comment:
I think this stage should be described in the docstring of refit, which is currently very short.

Reply:
@jpaillard

Reply:
@jpaillard